src/test/ui/param-bounds-ignored.stderr

```diff
@@ -1,18 +1,92 @@
-warning[E0122]: generic bounds are ignored in type aliases
-  --> $DIR/param-bounds-ignored.rs:15:1
+warning: bounds on generic parameters are ignored in type aliases
+  --> $DIR/param-bounds-ignored.rs:16:1
    |
-15 | type SVec<T: Send> = Vec<T>;
+16 | type SVec<T: Send> = Vec<T>;
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+   = note: #[warn(ignored_generic_bounds)] on by default
 
-warning[E0122]: generic bounds are ignored in type aliases
-  --> $DIR/param-bounds-ignored.rs:16:1
+warning: bounds on generic parameters are ignored in type aliases
+  --> $DIR/param-bounds-ignored.rs:17:1
    |
-16 | type VVec<'b, 'a: 'b> = Vec<&'a i32>;
+17 | type VVec<'b, 'a: 'b> = Vec<&'a i32>;
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-warning[E0122]: generic bounds are ignored in type aliases
-  --> $DIR/param-bounds-ignored.rs:17:1
+warning: bounds on generic parameters are ignored in type aliases
+  --> $DIR/param-bounds-ignored.rs:18:1
    |
-17 | type WVec<'b, T: 'b> = Vec<T>;
+18 | type WVec<'b, T: 'b> = Vec<T>;
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+warning: where clauses are ignored in trait bounds
+  --> $DIR/param-bounds-ignored.rs:19:1
+   |
+19 | type W2Vec<'b, T> where T: 'b = Vec<T>;
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+warning: bounds on generic parameters are ignored in higher-ranked function types (i.e., `for`)
+  --> $DIR/param-bounds-ignored.rs:39:8
+   |
+39 |     f: for<'xa, 'xb: 'xa> fn(&'xa i32, &'xb i32) -> &'xa i32)
+   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
Spans could be more precise, AST has spans for generics and for the ignored bounds themselves.

Reply:
I'm now changing this so the span points at the first ignored bound. Is there a way to "combine" spans? I found no span for all the bounds together, just the generic (which is the wrong place) and individual bounds.